scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -969,6 +969,187 @@ cdef class TransformedDensityRejection(Method):
+    dist : object
+        An instance of a class with ``pdf`` method.
+
+        * ``pdf``: PDF of the distribution. The signature of the PDF is
+          expected to be: ``def pdf(self, x: float) -> float``. i.e.
+          the PDF should accept a Python float and
+          return a Python float. It doesn't need to integrate to 1 i.e.
+          the PDF doesn't need to be normalized. If not normalized, `pdf_area`
+          should be set to the area under the PDF.
+
+    mode : float, optional
+        (Exact) Mode of the distribution. When the mode is ``None``, a slow
+        numerical routine is used to approximate it. Default is ``None``.
+    pdf_area : float, optional
+        Area under the PDF. Optionally, an upper bound to the area under
+        the PDF can be passed at the cost of increased rejection constant.
+        Default is 1.
+    domain : list or tuple of length 2, optional
+        The support of the distribution.
+        Default is ``None``. When ``None``:
+
+        * If a ``support`` method is provided by the distribution object
+          `dist`, it is used to set the domain of the distribution.
+        * Otherwise the support is assumed to be :math:`(-\infty, \infty)`.
+
+    cdf_at_mode : float, optional
+        CDF at the mode. It can be given to increase the performance of the
+        algorithm. The rejection constant is halfed when CDF at mode is given.
+        Default is ``None``.
+    random_state : {None, int, `numpy.random.Generator`,
+                        `numpy.random.RandomState`}, optional
+
+        A NumPy random number generator or seed for the underlying NumPy random
+        number generator used to generate the stream of uniform random numbers.
+        If `random_state` is None (or `np.random`), the `numpy.random.RandomState`
+        singleton is used.
+        If `random_state` is an int, a new ``RandomState`` instance is used,
+        seeded with `random_state`.
+        If `random_state` is already a ``Generator`` or ``RandomState`` instance then
+        that instance is used.
+
+    References
+    ----------
+    .. [1] UNU.RAN reference manual, Section 5.3.16,
+           "SROU - Simple Ratio-of-Uniforms method",
+           http://statmath.wu.ac.at/software/unuran/doc/unuran.html#SROU
+    .. [2] Leydold, Josef. "A simple universal generator for continuous and
+           discrete univariate T-concave distributions." ACM Transactions on
+           Mathematical Software (TOMS) 27.1 (2001): 66-82
+    .. [3] Leydold, Josef. "Short universal generators via generalized ratio-of-uniforms
+           method." Mathematics of Computation 72.243 (2003): 1453-1471
+
+    Examples
+    --------
+    >>> from scipy.stats.sampling import SimpleRatioUniforms
+
+    Suppose we have the normal distribution:
+
+    >>> class StdNorm:
+    ...     def pdf(self, x):
+    ...         return np.exp(-0.5 * x**2)
+
+    Notice that the PDF doesn't integrate to 1. We can either pass the exact
```

Comment:
I think we (or I :) should try to come up with another example: the typical situation to use SROU is when you need a fast setup, e.g., you have a distribution with a shape parameter a and you want to draw one 10 rvs for 1000 different a. Then the setup cost of the HINV and PINV will dominate the runtime, say 1000 * 0.1 seconds. SROU should be much faster in that case

Reply:
added the example.